### PR TITLE
MINT-197 ModalStack, MINT-222, MINT-235 Balances & Tokens, MINT-200 Enzyme, MINT-203 Exchange

### PR DESCRIPTION
**Pull request by ipavlenko**
Ready to merge after review

MINT-197 ModalStack
MINT-222, MINT-235 Balances & Tokens
MINT-200 Enzyme
MINT-203 Exchange page

### CODE REVIEW

File: src/components/dashboard/OrdersTable.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        <div styleName="footer">
+          <RaisedButton label="All Offers" primary />
+        </div>
+      </div>
+    )
+  }
+
+  renderRow(order, index) {
+
+    let [min1, min2] = ('' + order.min.toFixed(2)).split('.')
```
Comment (dkchv) [inline src/components/dashboard/OrdersTable.jsx]: const by default

File: src/components/dialogs/AddCurrencyDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    tokens: PropTypes.object,
+    isTokensLoaded: PropTypes.bool,
+    handleClose: PropTypes.func,
+    handleSave: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props)
+
+    let tokens = this.props.tokens.entrySeq().toArray()
```
Comment (dkchv) [inline src/components/dialogs/AddCurrencyDialog.jsx]: const by default

File: src/components/dialogs/AddCurrencyDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    this.state = {
+      items: tokens.map(([name, token]) => ({
+        selected: this.props.profile.tokens().contains(name),
+        token: token,
```
Comment (dkchv) [inline src/components/dialogs/AddCurrencyDialog.jsx]: use shortcuts

File: src/components/dialogs/AddCurrencyDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            </div>
+          </div>
+        </ModalDialog>
+      </CSSTransitionGroup>
+    )
+  }
+}
+
+function mapStateToProps (state) {
+  let session = state.get('session')
```
Comment (dkchv) [inline src/components/dialogs/AddCurrencyDialog.jsx]: const. Review all code

File: src/components/dialogs/AlertDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+export class AlertDialog extends React.Component {
+
+  render() {
+
+    return (
+      <Dialog
+          title={this.props.title}
+          actions={this.renderActions()}
+          modal={false}
+          open={true}
```
Comment (dkchv) [inline src/components/dialogs/AlertDialog.jsx]: true by default

File: src/components/dialogs/BuyTokensDialog.jsx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+                    </div>
+                  </div>
+                </div>
+                <div styleName="col2">
+                  <div className="ByTokensDialog__icons">
+                    <div className="row">
+                      <div className="col-xs-1">
+                        <div className="icon">
+                          <div className="content" style={{
+                            background: `#05326a url(${icons.time}) no-repeat center center`
```
Comment (dkchv) [inline src/components/dialogs/BuyTokensDialog.jsx]: can't use url in sass?

File: src/layouts/partials/InfoPartial.jsx [hunk 7]
```diff
@@ -1,35 +1,69 @@
...
+
+  static propTypes = {
+    account: PropTypes.string,
+    profile: PropTypes.object,
+    tokens: PropTypes.object,
+    isTokensLoaded: PropTypes.bool,
+    addCurrency: PropTypes.func
+  }
 
   constructor(props) {
```
Comment (dkchv) [inline src/layouts/partials/InfoPartial.jsx]: useless

File: src/models/ProfileModel.js [hunk 8]
```diff
@@ -1,12 +1,22 @@
...
   email: null,
-  company: null
+  company: null,
+  tokens: new Immutable.Set(),
 }) {
+
+  constructor (data = {}) {
+    super({
+      ...data,
+      tokens: new Immutable.Set(data ? data.tokens : undefined)
```
Comment (dkchv) [inline src/models/ProfileModel.js]: useless, data is always exists, see `data = {}`

File: src/pages/lib/DashboardPage.jsx [hunk 9]
```diff
@@ -1,12 +1,26 @@
...
 
   render() {
     return (
-      <div styleName={styles.root}>
-        <h3>DashboardPage</h3>
+      <div styleName="root">
+        <Partials.BrandPartial />
+        <CSSTransitionGroup
+          transitionName="transition-opacity"
+          transitionAppear={true}
```
Comment (dkchv) [inline src/pages/lib/DashboardPage.jsx]: by default

File: src/redux/modals/actions.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+export const MODALS_POP = 'modals/POP'
+export const MODALS_CLEAR = 'modals/CLEAR'
+
+export const modalsPush = ({ component, props }) =>
+  async (dispatch) => {
+    return dispatch({ type: MODALS_PUSH, component, props })
+  }
+
+export const modalsReplace = ({ component, props }) =>
+  async (dispatch) => {
```
Comment (dkchv) [inline src/redux/modals/actions.js]: remove `async` in these actions

File: src/redux/modals/actions.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    return dispatch({ type: MODALS_POP })
+  }
+
+export const modalsClear = () =>
+  async (dispatch) => {
+    return dispatch({ type: MODALS_CLEAR })
+  }
+
+export const modalsClose = modalsPop
```
Comment (dkchv) [inline src/redux/modals/actions.js]: what for?

Comment (dkchv) [inline src/redux/modals/actions.js]: also, `modals` -> `modal` if you clone one

File: src/styles/themes/default.scss [hunk 12]
```diff
@@ -1,110 +1,110 @@
-@import "~@/styles/partials/variables";
+@import "~styles/partials/variables";
 
 @font-face {
```
Comment (dkchv) [inline src/styles/themes/default.scss]: remove fonts from 'default' file to 'fonts'-like and collect all font-relative declarations there

File: src/models/ProfileModel.js (hunk 8 above)
Comment (ipavlenko) [inline src/models/ProfileModel.js]: No it isn't
Sometimes unfortunately we have null, not undefined.
See IPFS.js#get and UserManagerDAO.getCBEList

won't fix

File: src/components/dialogs/BuyTokensDialog.jsx (hunk 6 above)
Comment (ipavlenko) [inline src/components/dialogs/BuyTokensDialog.jsx]: No
1) We will remove this code after we got an ability to upload icons and specify colors;
2) Icons will not be embedded, they will be provided (loaded), so there will be an image as object, not as a style.

Won't wix

File: src/redux/modals/actions.js (hunk 10 above)
Comment (ipavlenko) [inline src/redux/modals/actions.js]: Thanks

File: src/redux/modals/actions.js (hunk 11 above)
Comment (ipavlenko) [inline src/redux/modals/actions.js]: Just to made module API programmer-friendly

File: src/components/dashboard/OrdersTable.jsx (hunk 1 above)
Comment (ipavlenko) [inline src/components/dashboard/OrdersTable.jsx]: Done

File: src/components/dialogs/AddCurrencyDialog.jsx (hunk 2 above)
Comment (ipavlenko) [inline src/components/dialogs/AddCurrencyDialog.jsx]: Done

File: src/components/dialogs/AddCurrencyDialog.jsx (hunk 3 above)
Comment (ipavlenko) [inline src/components/dialogs/AddCurrencyDialog.jsx]: Done

File: src/components/dialogs/AddCurrencyDialog.jsx (hunk 4 above)
Comment (ipavlenko) [inline src/components/dialogs/AddCurrencyDialog.jsx]: Done

File: src/components/dialogs/AlertDialog.jsx (hunk 5 above)
Comment (ipavlenko) [inline src/components/dialogs/AlertDialog.jsx]: Done

File: src/layouts/partials/InfoPartial.jsx (hunk 7 above)
Comment (ipavlenko) [inline src/layouts/partials/InfoPartial.jsx]: Done

File: src/pages/lib/DashboardPage.jsx (hunk 9 above)
Comment (ipavlenko) [inline src/pages/lib/DashboardPage.jsx]: Done

File: src/styles/themes/default.scss (hunk 12 above)
Comment (ipavlenko) [inline src/styles/themes/default.scss]: Done

File: src/models/ProfileModel.js (hunk 8 above)
Comment (dkchv) [inline src/models/ProfileModel.js]: ok. Add todo in UserManagerDAO.getCBEList
to parse `const user = new ProfileModel(await this._ipfs(hash))` before create object and similar

Comment (ipavlenko) [inline src/models/ProfileModel.js]: I think there may be several places with this problem.
I will live a comment on ProfileModel itself

Done

**Pull request merged**